src/io.jl

```diff
@@ -95,23 +102,33 @@ function Base.write(io::IO, d::MRCData; compress=:none, unit_vsize=4096)
     data = parent(d)
     fswap = bswapfromh(h.machst)
     unit_vsize = div(unit_vsize, sizeof(T))
+    if isnothing(buffer)
+        buffer = Vector{T}(undef, unit_vsize)
+    end
+    buffer::Vector{T}
+    # If `buffer` was provided as a parameter then `unit_vsize` is redundant and
+    # we must make sure that it matches `buffer`.
+    unit_vsize = length(buffer)
     vlen = length(data)
     vrem = vlen % unit_vsize
     if vrem != 0
-        @inbounds @views sz += write(newio, fswap.(T.(data[1:vrem])))
+        @inbounds @views buffer[1:vrem] .= fswap.(T.(data[1:vrem]))
+        @inbounds @views sz += write(newio, buffer[1:vrem])
     end
     for i in (vrem + 1):unit_vsize:vlen
-        @inbounds @views sz += write(newio, fswap.(T.(data[i:(i + unit_vsize - 1)])))
+        @inbounds @views buffer .= fswap.(T.(data[i:i + unit_vsize - 1]))
+        @inbounds sz += write(newio, buffer)
     end
-    close(newio)
+    write(newio, TranscodingStreams.TOKEN_END)
+    flush(newio)
     return sz
 end
-function Base.write(fn::AbstractString, object::T; compress=:auto, unit_vsize=4096) where {T<:Union{MRCData}}
+function Base.write(fn::AbstractString, object::T; compress=:auto, kwargs...) where {T<:Union{MRCData}}
     if compress == :auto
         compress = checkextension(fn)
     end
     return open(fn; write=true) do io
-        return write(io, object; compress=compress, unit_vsize=unit_vsize)
+        return write(io, object; compress, kwargs...)
```

Comment:
We need to explicitly write `compress-compress` to support Julia versions pre v1.7. 
```suggestion
        return write(io, object; compress=compress, kwargs...)
```